python/cudf_polars/cudf_polars/dsl/expressions/literal.py

```diff
@@ -61,10 +59,13 @@ class LiteralColumn(Expr):
     _non_child = ("dtype", "value")
     value: pa.Array[Any]
 
-    def __init__(self, dtype: plc.DataType, value: pl.Series) -> None:
+    def __init__(self, dtype: plc.DataType, value: pl.Series | pa.Array) -> None:
         self.dtype = dtype
-        data = value.to_arrow()
-        self.value = data.cast(dtypes.downcast_arrow_lists(data.type))
+        if isinstance(value, pa.Array):
+            self.value = value
+        else:
+            data = value.to_arrow()
+            self.value = data.cast(dtypes.downcast_arrow_lists(data.type))
```

Comment:
Let's simplify things and move the conversion into `translate.py`.

Reply:
Ah, great point. That's much cleaner.